### PR TITLE
Admin screen, group page, login screen

### PR DESCRIPTION
**Pull request by tarasIV**
I added Login screen in login component and I modified groups component (for displaying active + inactive classes, editing classes and adding new class).

### CODE REVIEW

File: src/app/admin-panel/admin-panel.module.ts [hunk 1]
```diff
@@ -7,13 +7,33 @@ import { StudentsComponent } from './admin-panel/students/students.component';
 import { TeachersComponent } from './admin-panel/teachers/teachers.component';
 import { GroupsComponent } from './admin-panel/groups/groups.component';
 import { SubjectsComponent } from './admin-panel/subjects/subjects.component';
+import {TableModule} from 'primeng/table';
```
Comment (dmitryy90) [inline src/app/admin-panel/admin-panel.module.ts]: merge all imports

File: src/app/admin-panel/admin-panel/groups/groups.component.html [hunk 2]
```diff
@@ -1,3 +1,78 @@
-<p>
-  groups works!
-</p>
+<div class="groups-container">
+    <p-table  [columns]="cols" [value]="activeGroups">
+        <ng-template pTemplate="caption">
+            Список класів
```
Comment (dmitryy90) [inline src/app/admin-panel/admin-panel/groups/groups.component.html]: indentation

File: src/app/admin-panel/admin-panel/groups/groups.component.html [hunk 3]
```diff
@@ -1,3 +1,78 @@
...
+
+
+<div>
+    <p-button (click)="showDialog()" label="Додати клас" styleClass="ui-button-success"></p-button>
+</div>
+
+</div>
+    <p-dialog header="Редагувати клас"  [(visible)]="showEditDialog" [modal]="true" [responsive]="true" [width]="450" [minWidth]="200" [minY]="70" 
+        [maximizable]="true" [baseZIndex]="10000">                   
+
```
Comment (dmitryy90) [inline src/app/admin-panel/admin-panel/groups/groups.component.html]: indentation

File: src/app/admin-panel/admin-panel/groups/groups.component.ts [hunk 4]
```diff
@@ -1,15 +1,95 @@
...
+                        this.groups = data;
+                        this.filterGroups();
+                      });
+        
+    this.cols = [
+        
+      { field: 'className', header: 'Клас' },
+      { field: 'classYear', header: 'Рік'}
+    ]                 
+    // this.groups = [ { id: 1, class: '5', year: 2018, description:"kj", isActive:true },
```
Comment (dmitryy90) [inline src/app/admin-panel/admin-panel/groups/groups.component.ts]: do we need this?

File: src/app/login/login.module.ts [hunk 5]
```diff
@@ -1,13 +1,23 @@
...
+import {ToolbarModule} from 'primeng/toolbar';
 
 import { LoginRoutingModule } from './login-routing.module';
 import { LoginComponent } from './login/login.component';
 
 @NgModule({
   imports: [
     CommonModule,
-    LoginRoutingModule
+		LoginRoutingModule,
```
Comment (dmitryy90) [inline src/app/login/login.module.ts]: indentation

File: src/app/login/login/login.component.html [hunk 6]
```diff
@@ -1,3 +1,25 @@
-<p>
-  login works!
-</p>
+<body>
+	<p-card>
+    <div class="container">
+   		<h1> Вхід до системи</h1>
```
Comment (dmitryy90) [inline src/app/login/login/login.component.html]: indentation

File: src/app/services/admingroups.service.ts [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+@Injectable()
+export class AdmingroupsService {
+    
+private ClassesUrl = 'https://fierce-shore-32592.herokuapp.com/classes';
+  
+constructor(private http: HttpClient) { }
+
+public getClasses(): Observable<Group[]> {
+  //TODO: remove this spike and use sigle method to load all the classes
```
Comment (dmitryy90) [inline src/app/services/admingroups.service.ts]: do we need commented code?

File: src/app/admin-panel/admin-panel/admin-panel.component.html [hunk 8]
```diff
@@ -7,4 +7,7 @@
     <div class="p-col-10">
       <router-outlet></router-outlet>
     </div> 
-  </div>
\ No newline at end of file
+	</div>
+
```
Comment (dmitryy90) [inline src/app/admin-panel/admin-panel/admin-panel.component.html]: what this place is reserved for?

File: src/app/admin-panel/admin-panel/groups/groups.component.html [hunk 9]
```diff
@@ -1,3 +1,77 @@
-<p>
-  groups works!
-</p>
+<div class="groups-container">
+	<p-table  [columns]="cols" [value]="activeGroups">
```
Comment (dmitryy90) [inline src/app/admin-panel/admin-panel/groups/groups.component.html]: please format file

File: src/app/admin-panel/admin.guard.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+			if (this._authService.loggedIn()) {
+				return true; 
+			}
+			this.router.navigate(['/login/']);
+			return false;		
+		}
+				
+}
+		
+
```
Comment (dmitryy90) [inline src/app/admin-panel/admin.guard.ts]: what this place is reserved for?

File: src/app/app-routing.module.ts [hunk 11]
```diff
@@ -1,14 +1,20 @@
...
-    path: 'login',
+		path: 'login',
+		canLoad: [LoginGuard],
     loadChildren: './login/login.module#LoginModule'
   },
   {
-    path: 'shell',
-    loadChildren: './shell/shell.module#ShellModule'
+		path: 'shell',
+		canLoad: [AdminGuard],
```
Comment (dmitryy90) [inline src/app/app-routing.module.ts]: admin gurard should be assigned to admin-panel only

File: src/app/admin-panel/admin.guard.ts [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+import { CanLoad, Route, Router } from '@angular/router';
+import { AuthenticationService } from '../services/authentication.service';
+
+@Injectable()
+export class AdminGuard implements CanLoad {
+	
+	constructor(private _authService: AuthenticationService,
+							private router : Router) { }
+		
+		canLoad(route: Route): boolean {
```
Comment (dmitryy90) [inline src/app/admin-panel/admin.guard.ts]: where the role is handled?

File: src/app/login/login/login.component.ts [hunk 13]
```diff
@@ -1,15 +1,44 @@
...
+	userName: string;
+	password: string;
+	errorMessage: string;
 
+  ngOnInit() {}
+	Login() {
+		
+		this.authService.login(this.userName, this.password)
+			.subscribe((result) => {
+					this.errorMessage = null;
```
Comment (dmitryy90) [inline src/app/login/login/login.component.ts]: indentation

File: src/app/login/login/login.guard.ts [hunk 14]
```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { CanLoad, Route, Router } from '@angular/router';
+import { AuthenticationService } from '../../services/authentication.service';
+
+@Injectable()
+export class LoginGuard implements CanLoad {
+	
+	constructor(private _authService: AuthenticationService,
```
Comment (dmitryy90) [inline src/app/login/login/login.guard.ts]: please format file

File: src/app/models/groupModel.ts [hunk 15]
```diff
@@ -0,0 +1,15 @@
+export class Group{
```
Comment (dmitryy90) [inline src/app/models/groupModel.ts]: rename group.model

File: src/app/services/authentication.service.ts [hunk 16]
```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthenticationService {
+	public Url:string="https://fierce-shore-32592.herokuapp.com/signin";
+
```
Comment (dmitryy90) [inline src/app/services/authentication.service.ts]: format file

File: src/app/services/token-interceptor.service.ts [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+import { AuthenticationService} from './authentication.service'
+import { Router } from '@angular/router';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TokenInterceptorService implements HttpInterceptor{
+
+	constructor(private authService: AuthenticationService,
```
Comment (dmitryy90) [inline src/app/services/token-interceptor.service.ts]: format file

Comment (tarasIV) [reply src/app/services/token-interceptor.service.ts]: Fixed

File: src/app/services/authentication.service.ts (hunk 16 above)
Comment (tarasIV) [reply src/app/services/authentication.service.ts]: Fixed

File: src/app/models/groupModel.ts (hunk 15 above)
Comment (tarasIV) [reply src/app/models/groupModel.ts]: Renamed

File: src/app/login/login/login.guard.ts (hunk 14 above)
Comment (tarasIV) [reply src/app/login/login/login.guard.ts]: Fixed

File: src/app/login/login/login.component.ts (hunk 13 above)
Comment (tarasIV) [reply src/app/login/login/login.component.ts]: Fixed

File: src/app/app-routing.module.ts (hunk 11 above)
Comment (tarasIV) [reply src/app/app-routing.module.ts]: Deleted

File: src/app/admin-panel/admin.guard.ts (hunk 10 above)
Comment (tarasIV) [reply src/app/admin-panel/admin.guard.ts]: Fixed

File: src/app/admin-panel/admin-panel/groups/groups.component.html (hunk 9 above)
Comment (tarasIV) [reply src/app/admin-panel/admin-panel/groups/groups.component.html]: Fixed

File: src/app/admin-panel/admin-panel/admin-panel.component.html (hunk 8 above)
Comment (tarasIV) [reply src/app/admin-panel/admin-panel/admin-panel.component.html]: Fixed.

**Comment by tarasIV** [comment]
Fixed conflicts in shell.

**Comment by tarasIV** [comment]
Fixed conflicts.

**Pull request merged**